lib/msf/ui/console/command_dispatcher/db.rb

```diff
@@ -275,7 +297,34 @@ def process_deleted_workspaces(deleted_workspaces, starting_ws)
 
   def cmd_workspace_tabs(str, words)
     return [] unless active?
-    framework.db.workspaces.map { |s| s.name } if (words & ['-a','--add']).empty?
+    framework.db.workspaces.map(&:name) if (words & ['-a','--add']).empty?
+  end
+
+  #
+  # Tab completion for the hosts command
+  #
+  # @param str [String] the string currently being typed before tab was hit
+  # @param words [Array<String>] the previously completed words on the command line.  words is always
+  # at least 1 when tab completion has reached this stage since the command itself has been completed
+
+  def cmd_hosts_tabs(str, words)
+    if words.length == 1
+      return @@hosts_opts.option_keys.select { |opt| opt.start_with?(str) }
+    end
+
+    case words[-1]
+    when '-d', '--delete'
+      # Might be a good idea to limit this to only 10 or so hosts
+      return framework.db.hosts[0..4].map(&:address) + ['...']
+    when '-c', '--columns', '-C', '--columns-until-restart'
+      # Should we get some default columns?
+      return ['address', 'arch', 'comm', 'comments', '...']
+    when '-O', '--order'
+      # Return some sample column ids
+      return ['1', '2', '3', '4', '...']
```

Comment:
In the future sorting by column names might be a better UX here, but for now let's not tab complete `...` to the users:

```suggestion
      return []
```